dev/io.openliberty.io.smallrye.reactive.messaging-provider4/src/io/openliberty/microprofile/reactive/messaging30/internal/OLReactiveMessaging30Extension.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.microprofile.reactive.messaging30.internal;
+
+import java.lang.annotation.Annotation;
+
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.spi.Connector;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+
+import com.ibm.ws.cdi.CDIServiceUtils;
+import com.ibm.ws.cdi.extension.WebSphereCDIExtension;
+
+import io.smallrye.reactive.messaging.providers.MediatorFactory;
+import io.smallrye.reactive.messaging.providers.OutgoingInterceptorDecorator;
+import io.smallrye.reactive.messaging.providers.connectors.WorkerPoolRegistry;
+import io.smallrye.reactive.messaging.providers.extension.ChannelProducer;
+import io.smallrye.reactive.messaging.providers.extension.EmitterFactoryImpl;
+import io.smallrye.reactive.messaging.providers.extension.HealthCenter;
+import io.smallrye.reactive.messaging.providers.extension.LegacyEmitterFactoryImpl;
+import io.smallrye.reactive.messaging.providers.extension.MediatorManager;
+import io.smallrye.reactive.messaging.providers.extension.MutinyEmitterFactoryImpl;
+import io.smallrye.reactive.messaging.providers.extension.ReactiveMessagingExtension;
+import io.smallrye.reactive.messaging.providers.impl.ConfiguredChannelFactory;
+import io.smallrye.reactive.messaging.providers.impl.ConnectorFactories;
+import io.smallrye.reactive.messaging.providers.impl.InternalChannelRegistry;
+import io.smallrye.reactive.messaging.providers.wiring.Wiring;
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.spi.AnnotatedType;
+import jakarta.enterprise.inject.spi.BeanManager;
+import jakarta.enterprise.inject.spi.BeforeBeanDiscovery;
+import jakarta.enterprise.inject.spi.Extension;
+
+@Component(service = WebSphereCDIExtension.class, configurationPolicy = ConfigurationPolicy.IGNORE, property = { "service.vendor=IBM", "application.bdas.visible=true" })
```

Comment:
https://ibm-cloud.slack.com/archives/C31DW78RG/p1633010438267800
Thomas Watson:
I've been touching various service components in Open Liberty and often times I convert a bnd.bnd instruction for Service-Component to use annotations instead.  Most all the time I do that I leave the service property service.vendor=IBM out of the annotation for @Component.  I see no value to this service property.  It certainly seems strange at least on an open source project.  Is there any reason we need to keep this property on all our service components?




2 replies


Alasdair Nottingham
  [2 years ago](https://ibm-cloud.slack.com/archives/C31DW78RG/p1633010843267900?thread_ts=1633010438.267800&cid=C31DW78RG)
I don't think we need to keep it


Thomas Watson
  [2 years ago](https://ibm-cloud.slack.com/archives/C31DW78RG/p1633012652268100?thread_ts=1633010438.267800&cid=C31DW78RG)
ok good

Reply:
OK, fair point ... I might do a separate PR to remove it from here and elsewhere.